x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/jwt/JwtRealmSettings.java

```diff
@@ -0,0 +1,310 @@
+
+    // Default values and min/max constraints
+
+    private static final TimeValue DEFAULT_ALLOWED_CLOCK_SKEW = TimeValue.timeValueSeconds(60);
+    private static final List<String> DEFAULT_ALLOWED_SIGNATURE_ALGORITHMS = Collections.singletonList("RS256");
+    private static final List<String> DEFAULT_ALLOWED_AUDIENCES = Collections.emptyList();
+    private static final boolean DEFAULT_POPULATE_USER_METADATA = true;
+    private static final String DEFAULT_CLIENT_AUTHENTICATION_TYPE = SUPPORTED_CLIENT_AUTHENTICATION_TYPE.get(0);
+    private static final TimeValue DEFAULT_CACHE_TTL = TimeValue.timeValueMinutes(20);
+    private static final int DEFAULT_CACHE_MAX_USERS = 100_000;
+    private static final int MIN_CACHE_MAX_USERS = 0;
+    private static final TimeValue DEFAULT_HTTP_CONNECT_TIMEOUT = TimeValue.timeValueSeconds(5);
+    private static final TimeValue DEFAULT_HTTP_CONNECTION_READ_TIMEOUT = TimeValue.timeValueSeconds(5);
+    private static final TimeValue DEFAULT_HTTP_SOCKET_TIMEOUT = TimeValue.timeValueSeconds(5);
+    private static final int DEFAULT_HTTP_MAX_CONNECTIONS = 200;
+    private static final int MIN_HTTP_MAX_CONNECTIONS = 0;
+    private static final int DEFAULT_HTTP_MAX_ENDPOINT_CONNECTIONS = 200;
+    private static final int MIN_HTTP_MAX_ENDPOINT_CONNECTIONS = 0;
+    private static final int DEFAULT_HTTP_PROXY_PORT = 80;
+    private static final int MIN_HTTP_PROXY_PORT = 1;
+    private static final int MAX_HTTP_PROXY_PORT = 65535;
+    private static final String DEFAULT_HTTP_PROXY_SCHEME = "http";
+
+    // All settings
+
+    public static Set<Setting.AffixSetting<?>> getSettings() {
+        final Set<Setting.AffixSetting<?>> set = Sets.newHashSet();
+        // Standard realm settings: order, enabled
+        set.addAll(RealmSettings.getStandardSettings(TYPE));
+        // JWT Issuer settings
+        set.addAll(List.of(ALLOWED_ISSUER, ALLOWED_SIGNATURE_ALGORITHMS, ALLOWED_CLOCK_SKEW, JWKSET_PATH));
+        // JWT Audience settings
+        set.addAll(List.of(ALLOWED_AUDIENCES));
+        // JWT End-user settings
+        set.addAll(
+            List.of(
+                PRINCIPAL_CLAIM.getClaim(),
+                PRINCIPAL_CLAIM.getPattern(),
+                GROUPS_CLAIM.getClaim(),
+                GROUPS_CLAIM.getPattern(),
+                POPULATE_USER_METADATA
+            )
+        );
+        // Client TLS settings for incoming connections (subset of SSLConfigurationSettings)
+        set.addAll(
+            List.of(
+                CLIENT_AUTHENTICATION_TYPE,
+                CLIENT_AUTHENTICATION_TRUSTSTORE_PATH,
+                CLIENT_AUTHENTICATION_TRUSTSTORE_PASSWORD,
+                CLIENT_AUTHENTICATION_LEGACY_TRUSTSTORE_PASSWORD,
+                CLIENT_AUTHENTICATION_TRUSTSTORE_ALGORITHM,
+                CLIENT_AUTHENTICATION_TRUSTSTORE_TYPE,
+                CLIENT_AUTHENTICATION_CERT_AUTH_PATH
+            )
+        );
```

Comment:
I suggest we add this when we add support for client tls authentication, I presume this will be in a different PR than the one that introduces the realm